src/main/java/org/isf/examination/rest/ExaminationController.java

```diff
@@ -268,4 +199,50 @@ public ResponseEntity<List<PatientExaminationDTO>> getByPatientId(@PathVariable
+        }
+        if (newPatientExamination.getPex_ap_min() != null && newPatientExamination.getPex_ap_max() != null) {
+        	if (newPatientExamination.getPex_ap_min() > newPatientExamination.getPex_ap_max()) {
+            	throw new OHAPIException(new OHExceptionMessage("The minimum blood pressure must be lower than the maximum blood pressure"));
+            }
+        }
+        if (newPatientExamination.getPex_hr() != null) {
+        	if (newPatientExamination.getPex_hr() < ExaminationParameters.HR_MIN || newPatientExamination.getPex_hr() > ExaminationParameters.HR_MAX) {
+            	throw new OHAPIException(new OHExceptionMessage("Heart rate should be between "+ExaminationParameters.HR_MIN +" and "+ExaminationParameters.HR_MAX));
+            }
+        }
+        if (newPatientExamination.getPex_temp() != null) {
+        	if (newPatientExamination.getPex_temp() < ExaminationParameters.TEMP_MIN || newPatientExamination.getPex_temp() > ExaminationParameters.TEMP_MAX) {
+             	throw new OHAPIException(new OHExceptionMessage("The temperature should be between "+ExaminationParameters.TEMP_MIN +" and "+ExaminationParameters.TEMP_MAX));
+            }
+        }
+	    if (newPatientExamination.getPex_sat() != null) {
+	    	if (newPatientExamination.getPex_sat() < ExaminationParameters.SAT_MIN || newPatientExamination.getPex_sat() > ExaminationParameters.SAT_MAX) {
+	         	throw new OHAPIException(new OHExceptionMessage("The saturation should be between "+ExaminationParameters.SAT_MIN+" and "+ExaminationParameters.SAT_MAX));
+	        } 
+	    }
+        if (newPatientExamination.getPex_hgt() != null) {
+        	if (newPatientExamination.getPex_hgt() < ExaminationParameters.HGT_MIN || newPatientExamination.getPex_hgt() > ExaminationParameters.HGT_MAX) {
+            	throw new OHAPIException(new OHExceptionMessage("HGT should be between "+ExaminationParameters.HGT_MIN+" and "+ExaminationParameters.HGT_MAX));
+            }
+        }
+        if (newPatientExamination.getPex_rr() != null) {
+        	if (newPatientExamination.getPex_rr() < ExaminationParameters.RR_MIN || newPatientExamination.getPex_rr() > ExaminationParameters.RR_MAX) {
+            	throw new OHAPIException(new OHExceptionMessage("Respiratory rate should be between "+ExaminationParameters.RR_MIN+" and "+ExaminationParameters.RR_MAX));
+            }
+        }
+        if (newPatientExamination.getPex_diuresis() != null) {
+        	if (newPatientExamination.getPex_diuresis() < ExaminationParameters.DIURESIS_MIN || newPatientExamination.getPex_diuresis() > ExaminationParameters.DIURESIS_MAX) {
+            	throw new OHAPIException(new OHExceptionMessage("Diuresis should be between "+ExaminationParameters.DIURESIS_MIN+" and "+ExaminationParameters.DIURESIS_MAX));
+            }
+        }
+    }
```

Comment:
Now what is missing is that if **Diuresis**, **Bowel** and **Auscultation** are provided, they should be contained in their respective enumerations

Reply:
Found out that because of the casting in DTO, values different from enum are not accepted by default, so these check are not even needed, do you think we can keep them?

```
[27/oct./2023 16:19:54] [:] WARN - Resolved [org.springframework.http.converter.HttpMessageNotReadableException: JSON parse error: Cannot deserialize value of type `org.isf.examination.dto.Ausculation` from String "ciao": not one of the values accepted for Enum class: [bronchial, normal, rhonchi, crackles, wheezes, stridor]; nested exception is com.fasterxml.jackson.databind.exc.InvalidFormatException: Cannot deserialize value of type `org.isf.examination.dto.Ausculation` from String "ciao": not one of the values accepted for Enum class: [bronchial, normal, rhonchi, crackles, wheezes, stridor]<EOL> at [Source: (org.springframework.util.StreamUtils$NonClosingInputStream); line: 7, column: 23] (through reference chain: org.isf.examination.dto.PatientExaminationDTO["pex_auscultation"])]
```

Nevertheless, the enumerations **Diuresis**, **Bowel** and **Auscultation** should be shared within core, which means to move them into core and let API (and GUI) use them. Do you think we can do this now or in another PR?

Reply:
yes I can remove it on api an add it on the core on this PR

Reply:
ok, let's try